### ISSUE TITLE
Broken `--enable-suppress-external-warning` for Apple Clang 15 on `x86_64`

### ISSUE THREAD

**Issue by hebasto** [bitcoin/bitcoin#29174]
Apple Clang 15 does not support `-Xclang -internal-isystem` anymore. On both `x86_64` and `arm64` platforms, the following check: https://github.com/bitcoin/bitcoin/blob/65c05db660b2ca1d0076b0d8573a6760b3228068/configure.ac#L742-L744
 fails with `error: unknown argument: '-internal-isystem/usr/local/include'`.

It makes the `--enable-suppress-external-warnings` option not working for dependency packages installed by Homebrew and linked to `/usr/local`, which happens for all non-keg packages on `x86_64`. There is no such an effect on `arm64`.

For a related discussion refer to https://github.com/bitcoin/bitcoin/pull/29165#discussion_r1440273658.

**Comment by hebasto** [comment]
Here are more details:

1.
```
% clang -cc1 --help | grep -A 1 internal-isystem
  -internal-isystem <directory>
                          Add directory to the internal system include search path; these are assumed to not be user-provided and are used to model system and standard headers' paths.
```


2.
> Is the issue here just the concatenation? Maybe that worked previously, but no-longer does? What happens if you change `-Xclang -internal-isystem/usr/local/include` to `-Xclang -internal-isystem /usr/local/include`?

```
% clang++ -Xclang -internal-isystem /usr/local/include hello.cpp       
ld: file cannot be mmap()ed, errno=22 path=/usr/local/include in '/usr/local/include'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by hebasto** [comment]
According to [Wikipedia](https://en.wikipedia.org/wiki/Xcode#Xcode_15.0_-_(since_visionOS_support)_2), the Apple clang version 15.0.0 (clang-1500.0.40.1) corresponds to LLVM clang 16.0.0.

On Ubuntu 23.10:
```
$ clang++-16 -Xclang -internal-isystem/usr/local/include test.cpp 
error: unknown argument: '-internal-isystem/usr/local/include'
```

However,
```
$ clang++-15 -Xclang -internal-isystem/usr/local/include test.cpp  # Works.
```

**Comment by hebasto** [comment]
Fixed in #29195.
